tools/ci/pr_preview.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/env python
+
+# The service provided by this script is not critical, but it shares a GitHub
+# API request quota with critical services. For this reason, all requests to
+# the GitHub API are preceded by a "guard" which verifies that the subsequent
+# request will not deplete the shared quota.
+#
+# In effect, this script will fail rather than interfere with the operation of
+# critical services.
+
+import argparse
+import json
+import logging
+import os
+import subprocess
+import time
+
+import requests
+
+# The ratio of "requests remaining" to "total request quota" below which this
+# script should refuse to interact with the GitHub.com API
+API_RATE_LIMIT_THRESHOLD = 0.2
+# The GitHub Pull Request label which indicates that a Pull Request is expected
+# to be actively mirrored by the preview server
+LABEL = 'safelisted-for-preview'
```

Comment:
```suggestion
LABEL = 'safe for preview'
```

Mostly because it's shorter. It seems like most labels either match directory names, are on the form "foo:bar-baz" or "human readable string". I didn't count though, and did see exceptions.

Reply:
It's more accurate, as well, since adding the label *is* what makes a pull request "safelisted."